### PR TITLE
Fix course theme comment styles

### LINKED ISSUE #6977: Fix comment styles
The comment styles on some post types look better than on others, but none of them match the design on [desktop](https://www.figma.com/file/nf4oV51c1JBxJKXshMe7cd/New-Sensei-Theme?type=design&node-id=20020065-1078&t=w4heqOhmBBU9luEY-0) or on [mobile](https://www.figma.com/file/nf4oV51c1JBxJKXshMe7cd/New-Sensei-Theme?type=design&node-id=20020126-4278&t=w4heqOhmBBU9luEY-0). Lesson comments (when Learning Mode is disabled) are especially bad and comments are actually duplicated there.

We should update page, post and lesson comments so that they match the designs.

### Page
screencapture-senseitheme-local-about-page-with-comments-2023-06-21-10_57_14

### Post
Screenshot 2023-06-21 at 11 02 51 AM

### Lesson
screencapture-senseitheme-local-lesson-biggest-trap-117-comment-page-1-2023-06-21-11_08_46

### Design
comments

Be sure to check the design when logged in and when not logged in, as different form fields are available in both scenarios.

### PR DESCRIPTION
**Pull request by Imran92**
#### Changes proposed in this Pull Request:

Fixes discrepancies in the comment thread and reply form styles among different templates, like posts, page, lesson, and optional templates.

### Testing instructions

1. Enable the Course theme
2. Enable Sensei
3. Create a page and a post, make sure to enable comments
4. Create a Course with a lesson in sensei, and enable comments for it too
5. Now visit the page and post from the frontend as a logged out user
6. Make a few comments
7. Make sure the thread and the reply forms look close to the design
8. Now open the post in the editor
9. Select each custom template from the right dropdown, like Single Column - Featured Image Only or Single Column - Title only, etc.
10. After selecting each of them, click on update and check from the frontend. Make sure the comments are not broken in any of them.
11. Now check them again as logged in user.
12. You won't see all the fields now, but make sure it doesn't appear broken.
13. Make sure to check for the lesson now too. Add a couple of comments, make sure the thread looks like the other ones from page and post type.

#### For now, disregard if you see the comment portion getting duplicated in lessons. It'll be fixed in a separate PR under a different issue. Only check for the comment portion at the bottom, not the top one.

We tried reusing the existing spacing and size presets to not introduce new ones. So in some places, the spacings are few pixels different, but they don't fall far from the design. Also, the design has some spacings that are not uniform, like the spacing between all the input fields or between the input fields and their labels, like the spacings between the email label and its input field and the comment label and its input field. We used a uniform spacing in these places.

image

### Screenshots

Desktop:
image

Mobile:
image

Design:
image

#### Related issue(s):

https://github.com/Automattic/themes/issues/6977

### CODE REVIEW

**Comment by donnapep** [comment]
When Learning Mode is disabled, the lessons page looks pretty terrible. Let me know if you think we should create a separate card for this:

Screenshot 2023-07-24 at 8 26 01 AM

**Comment by Imran92** [comment]
> For now, disregard if you see the comment portion getting duplicated in lessons. It'll be fixed in a separate PR under a different issue. Only check for the comment portion at the bottom, not the top one.

Hi @donnapep 👋 Thanks for taking a look. The one you are seeing is the screenshot is the comment section that's being automatically generated by sensei and not by the theme, so it's actually the 'duplicate' one that we see in the FSE themes. It's being worked on under this issue https://github.com/Automattic/sensei/issues/7037 already. It's also mentioned in the PR description above

Thanks!

File: course/patterns/comments.php [hunk 1]
```diff
@@ -8,42 +8,41 @@
 ?>
 
 <!-- wp:comments {"className":"wp-block-comments-query-loop "} -->
-<div class="wp-block-comments wp-block-comments-query-loop">
-<!-- wp:comments-title {"showPostTitle":false,"showCommentsCount":false,"level":3,"style":{"typography":{"fontStyle":"normal","fontWeight":"600"}},"fontSize":"large"} /-->
+<div class="wp-block-comments wp-block-comments-query-loop"><!-- wp:comments-title {"showPostTitle":false,"showCommentsCount":false,"level":3,"style":{"typography":{"fontStyle":"normal","fontWeight":"600","lineHeight":"1"},"spacing":{"padding":{"top":"0","right":"0","bottom":"40px","left":"0"},"margin":{"top":"0","right":"0","bottom":"0","left":"0"}}},"fontSize":"large"} /-->
```
Comment (donnapep) [inline course/patterns/comments.php]: Is this `40px` adding whitespace between 2 elements? If so, that should be margin instead of padding.

**Review by donnapep — COMMENTED** [review]
Here's some things I found while testing:

1. The size of the avatar for post comments (50 x 50) is different than it is for page and lesson comments (42 x 42). The post comments is closer to the design, which is 48 x 48 including the border. (But 50 x 50 is also 🆗 so long as they're all consistent.)
2. The space between the avatar and the username / date is also inconsistent (should be 20px for all).
3. There's extra whitespace between the line and the _Responses_ heading for posts. There's also a bit extra below (`40px` vs `36px` for pages / lessons):

Screenshot 2023-07-24 at 8 41 21 AM

As compared to pages or lessons, which has `36px` top and bottom:

Screenshot 2023-07-24 at 8 42 01 AM
In the design the top and bottom margins are `48px`, but again, I'm OK with `36px` as long as we're consistent across all post types.
4. Paged comments don't look too good. Although there's no design for it, could we move pagination closer to the comments and put the `80px` of bottom margin between it and "Leave a Reply"?

Screenshot 2023-07-24 at 9 01 49 AM

I haven't gone over all the details of the design with a fine-toothed comb, so please consider this review incomplete. Mostly I'm looking for the comment styles to be consistent between posts, pages and lessons and to generally look good. I'll go over it more closely once you've given it another pass. 🙂

**Comment by Imran92** [comment]
Hi @donnapep 👋 Thanks so much for checking!

1. 

> The size of the avatar for post comments (50 x 50) is different than it is for page and lesson comments (42 x 42). The post comments is closer to the design, which is 48 x 48 including the border. (But 50 x 50 is also 🆗 so long as they're all consistent.)

I couldn't reproduce this. Can you kindly check if your local environment is using this branch and there is no saved customization for any template from previous times? We are using one common pattern file and are using its reference from all other templates. So it should be consistent all over, in case there isn't targeted CSS for specific pattern or post type. I tried with different browsers, incognito mode etc. Here are some screenshots from my env
Page:
Page 48
Post:
Post 48
Lesson:
Lesson 48

2. 

> The space between the avatar and the username / date is also inconsistent (should be 20px for all).

Same as above, I couldn't reproduce this issue. It should be same because of using same pattern file. Here are my screenshots (I've used 20px column gap, the scale extension picks 22px). The dates also look the same in the screenshots
Page:
Page 22
Post:
Post 22
Lesson:
Lesson 22

3.

> There's extra whitespace between the line and the Responses heading for posts. There's also a bit extra below (40px vs 36px for pages / lessons):

Thanks! I've changed it here https://github.com/Automattic/themes/pull/7252/commits/7ade40da55c6454780abfc9c6979e3147b29bb76.

4. 

> Paged comments don't look too good. Although there's no design for it, could we move pagination closer to the comments and put the 80px of bottom margin between it and "Leave a Reply"?

Good suggestion. The pagination looked a bit out of place to me too. I've updated the spacings here https://github.com/Automattic/themes/pull/7252/commits/dafffcc548bcc2f333f04c4b503863f59c2ad67a

5. 

> Is this 40px adding whitespace between 2 elements? If so, that should be margin instead of padding.

Thanks for the good suggestion, updated here https://github.com/Automattic/themes/pull/7252/commits/dafffcc548bcc2f333f04c4b503863f59c2ad67a

Thanks again for taking a look @donnapep ! Please give it another check when you can :) !

File: course/theme.json [hunk 2]
```diff
@@ -137,6 +137,7 @@
 			"commentGap" : "5rem",
 			"commentGapSmall" : "10px",
 			"commentGapMedium" : "1.25rem",
+			"commentTopMargin": "clamp(2.5rem, 2.167rem + 1.111vw, 3rem)",
```
Comment (donnapep) [inline course/theme.json]: I think custom presets are meant to be used when we're using their values in multiple places. It doesn't appear that we're doing that here so I'd put it right in the CSS.

**Review by donnapep — COMMENTED** [review]
@Imran92 Thanks! 1 & 2 are resolved. It does look like I had a customization. 😳 

For number 3, I still see some extra whitespace. In the screenshot, the `.wp-block-group` element has 40px of margin, then there's a spacer that is `4rem`, and then `.wp-block-comments` also has 40px of margin.

Screenshot 2023-07-27 at 12 31 07 PM

The pagination isn't fixed for me either:

Screenshot 2023-07-27 at 12 33 57 PM

Everything else is looking great!

File: course/patterns/comments.php [hunk 3]
```diff
@@ -8,42 +8,41 @@
...
-<div class="wp-block-group"><!-- wp:avatar {"size":40,"style":{"border":{"radius":"24px","width":"1px"}}} /-->
+	<!-- wp:comments-title {"showPostTitle":false,"showCommentsCount":false,"level":3,"style":{"typography":{"fontStyle":"normal","fontWeight":"600","lineHeight":"1"},"spacing":{"padding":{"top":"0","right":"0","bottom":"0px","left":"0"},"margin":{"top":"0","right":"0","left":"0"}}},"fontSize":"large"} /-->
 
-<!-- wp:group {"style":{"spacing":{"blockGap":"0","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","orientation":"vertical"}} -->
-<div class="wp-block-group" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0"><!-- wp:comment-author-name /-->
+	<!-- wp:comment-template {"style":{"spacing":{"padding":{"top":"0","right":"0","bottom":"0","left":"0"},"margin":{"top":"0","right":"0","bottom":"0","left":"0"}}}} -->
+	<!-- wp:group {"style":{"spacing":{"padding":{"top":"0","right":"0","left":"0","bottom":"5rem"},"blockGap":"0"}},"layout":{"type":"flex","orientation":"vertical","flexWrap":"wrap"}} -->
```
Comment (donnapep) [inline course/patterns/comments.php]: Should this be margin instead of padding too?

**Comment by Imran92** [comment]
> @Imran92 Thanks! 1 & 2 are resolved. It does look like I had a customization. 😳

Whew! Glad it worked! ☺️ Thanks for finding out the customization.

> For number 3, I still see some extra whitespace. In the screenshot, the .wp-block-group element has 40px of margin, then there's a spacer that is 4rem, and then .wp-block-comments also has 40px of margin.

I couldn't reproduce this one. For me, all of page, post and lesson had 48px (the scale extension picks 49) above and below the Responses title. Can there be any issue like customization or caching here? Here are some screenshots from my environment. And just to make sure, I've tested with every template in case you had a different one. But still didn't find any issue

Page:
Screenshot 2023-07-27 at 10 51 45 PM

Post:
Screenshot 2023-07-27 at 10 52 18 PM

Lesson:
Screenshot 2023-07-27 at 10 53 07 PM

> The pagination isn't fixed for me either:

Fixed it here https://github.com/Automattic/themes/pull/7252/commits/a39ede88778a7dbb8f77b863eb679a21fa855136 ☺️  Should work now!

> I think custom presets are meant to be used when we're using their values in multiple places. It doesn't appear that we're doing that here so I'd put it right in the CSS.

Yap, you're right. Used the spacing directly in CSS here https://github.com/Automattic/themes/pull/7252/commits/a39ede88778a7dbb8f77b863eb679a21fa855136

> Should this be margin instead of padding too?

Changed it to margin here https://github.com/Automattic/themes/pull/7252/commits/ea225bedad4058b620f158d45bdfbc9c2d4cc95d

Thanks @donnapep ! Please take a look again when possible

**Comment by donnapep** [comment]
I figured out what the problem was (sync issue) and am no longer seeing the previous issues. 🙌🏻 

After doing a bit more extensive testing, I found one other issue when replying to threaded comments:

Screenshot 2023-07-31 at 8 48 39 AM

- We should style the _Cancel Reply_ link so that it looks the same as the _Edit_ link.
- Is it possible to make the text in the textbox `var(--wp--preset--color--foreground)` instead of black?
- There needs to be a bit more whitespace between the _Post Comment_ button and the following comment. `5rem` would match the current spacing between comments.

**Comment by donnapep** [comment]
Also, given the Themes team will end up deploying our changes along with any changes they have ready to ship as soon as we merge to `trunk`, I think we should consider putting these changes into a feature branch and, depending on how much we accomplish, maybe push a new release halfway through the project. Perhaps releasing just the theme, or also Sensei if there are any dependencies (which we should be sure to track). Thoughts?

File: course/style.css [hunk 4]
```diff
@@ -127,6 +127,61 @@ h1, h2, h3, h4, h5, h6 {
 /*
  * Comments
  */
+
+.wp-block-comments-query-loop .wp-block-comments-pagination {
+	margin-top: calc( var(--wp--custom--comment-gap) / 2 * -1 );
+	padding-bottom: var(--wp--custom--comment-gap);
```
Comment (donnapep) [inline course/style.css]:
```suggestion
	margin-bottom: var(--wp--custom--comment-gap);
```

File: course/patterns/comments.php [hunk 5]
```diff
@@ -8,42 +8,41 @@
...
 
-<!-- wp:group {"style":{"spacing":{"blockGap":"0","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","orientation":"vertical"}} -->
-<div class="wp-block-group" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0"><!-- wp:comment-author-name /-->
+	<!-- wp:comment-template {"style":{"spacing":{"padding":{"top":"0","right":"0","bottom":"0","left":"0"},"margin":{"top":"0","right":"0","bottom":"0","left":"0"}}}} -->
+	<!-- wp:group {"style":{"spacing":{"padding":{"top":"0","right":"0","left":"0","bottom":"0rem"},"blockGap":"0","margin":{"bottom":"5rem"}}},"layout":{"type":"flex","orientation":"vertical","flexWrap":"wrap"}} -->
+	<div class="wp-block-group" style="margin-bottom:5rem;padding-top:0;padding-right:0;padding-bottom:0rem;padding-left:0"><!-- wp:group {"style":{"spacing":{"blockGap":"20px","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"left","verticalAlignment":"top"}} -->
```
Comment (donnapep) [inline course/patterns/comments.php:19]:
```suggestion
	<div class="wp-block-group" style="margin-bottom:5rem;padding-top:0;padding-right:0;padding-bottom:0;padding-left:0"><!-- wp:group {"style":{"spacing":{"blockGap":"20px","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"left","verticalAlignment":"top"}} -->
```

File: course/patterns/comments.php [hunk 6]
```diff
@@ -8,42 +8,41 @@
...
-<div class="wp-block-group"><!-- wp:avatar {"size":40,"style":{"border":{"radius":"24px","width":"1px"}}} /-->
+	<!-- wp:comments-title {"showPostTitle":false,"showCommentsCount":false,"level":3,"style":{"typography":{"fontStyle":"normal","fontWeight":"600","lineHeight":"1"},"spacing":{"padding":{"top":"0","right":"0","bottom":"0px","left":"0"},"margin":{"top":"0","right":"0","left":"0"}}},"fontSize":"large"} /-->
 
-<!-- wp:group {"style":{"spacing":{"blockGap":"0","padding":{"top":"0","right":"0","bottom":"0","left":"0"}}},"layout":{"type":"flex","orientation":"vertical"}} -->
-<div class="wp-block-group" style="padding-top:0;padding-right:0;padding-bottom:0;padding-left:0"><!-- wp:comment-author-name /-->
+	<!-- wp:comment-template {"style":{"spacing":{"padding":{"top":"0","right":"0","bottom":"0","left":"0"},"margin":{"top":"0","right":"0","bottom":"0","left":"0"}}}} -->
+	<!-- wp:group {"style":{"spacing":{"padding":{"top":"0","right":"0","left":"0","bottom":"0rem"},"blockGap":"0","margin":{"bottom":"5rem"}}},"layout":{"type":"flex","orientation":"vertical","flexWrap":"wrap"}} -->
```
Comment (donnapep) [inline course/patterns/comments.php]: Can just be `0` instead of `0rem` for padding.

File: course/style.css [hunk 7]
```diff
@@ -127,6 +127,61 @@ h1, h2, h3, h4, h5, h6 {
...
+}
+
+.wp-block-comments-query-loop hr,
+.wp-block-comments-query-loop .wp-block-comments-title {
+	margin-bottom: clamp(2.5rem, 2.167rem + 1.111vw, 3rem);
+}
+
+.wp-block-post-comments-form .comment-form-cookies-consent input[type="checkbox"] {
```
Comment (donnapep) [inline course/style.css:149]: How can I test the cookies consent form?

Comment (Imran92) [reply course/style.css:149]: It is this checkbox on the bottom that you see when logged out 
image

**Comment by Imran92** [comment]
> I think we should consider putting these changes into a feature branch and, depending on how much we accomplish, maybe push a new release halfway through the project.

Makes sense to me, we can keep track of which version of the theme received the updates pertaining to the frontend improvement projects this way

**Comment by Imran92** [comment]
Hi @donnapep , Thanks for the comments!

> We should style the Cancel Reply link so that it looks the same as the Edit link.

Updated here https://github.com/Automattic/themes/pull/7252/commits/1eb1439a18027a33972c4874620deb4e62fd62c5

> Is it possible to make the text in the textbox var(--wp--preset--color--foreground) instead of black?

Updated the color here https://github.com/Automattic/themes/pull/7252/commits/82cca3ac0c5c4c5af4bfa04e935dd8a571c8766d

> There needs to be a bit more whitespace between the Post Comment button and the following comment. 5rem would match the current spacing between comments.

Added space here https://github.com/Automattic/themes/pull/7252/commits/57d40fc7f4d5dfb1b24d24c24c9b5cd9e6c5af59

> Can just be 0 instead of 0rem for padding.

Changed it here https://github.com/Automattic/themes/pull/7252/commits/deca37e0e1ef49ba026b1d5d3ba2ca84f3e67a08

Also, I want to ask your opinion about a style. When writing inside the comment reply form fields, the text size caused some eye strain for me because of being too small. Did you feel the same way? Maybe we can increase the font size a bit if you felt the same?

I've created a feature/frontend-improvements branch as per our discussion. I'll change the target branch from base to our feature branch for this PR.

Thanks again for checking. Please take another look when you can!

**Comment by donnapep** [comment]
> Also, I want to ask your opinion about a style. When writing inside the comment reply form fields, the text size caused some eye strain for me because of being too small. Did you feel the same way? Maybe we can increase the font size a bit if you felt the same?

Yup, I agree. The `x-small` size looked pretty good when I tried it. Maybe we could do it for all these reply form fields actually:

Screenshot 2023-08-02 at 10 43 16 AM

> I've created a feature/frontend-improvements branch as per our discussion. I'll change the target branch from base to our feature branch for this PR.

As per our 1:1 chat, we'll plan to release this one next week, so we can keep the branch at `trunk` and merge when we're ready to release.

**Comment by Imran92** [comment]
Hi @donnapep 👋 I've updated the comment text size here https://github.com/Automattic/themes/pull/7252/commits/635f6a7ac46e5e7367a0ad6168e83e77e592fbf7 to x-small, looks good to me too. Have also changed the target branch to trunk.

**Review by donnapep — APPROVED** [review]
Looks great! 🥳

**Pull request merged**